//
// System K - A framework for building trading applications and analysis
// Copyright (C) 2013 Timir Karia <tkaria@capitalkpartners.com>
//
// This file is part of System K.
//
// System K is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// System K is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with System K.  If not, see <http://www.gnu.org/licenses/>.
//

package capkproto;

import "capk_globals.proto";

message venue_configuration {
  optional sfixed32   venue_id                    = 1;
  optional string     mic_name                    = 2;
  optional string     order_interface_addr        = 3; 
  optional string     order_ping_addr             = 4; 
  optional string     market_data_broadcast_addr  = 5;
  optional string     logging_broadcast_addr      = 6;
  optional bool       use_synthetic_cancel_replace = 7;
}

message configuration {
  optional string     trade_serialization_addr    = 1;
  optional string     recovery_listener_addr      = 2;
  optional string     aggregated_bbo_book_addr    = 3;
  optional sfixed32   aggregated_bbo_book_id      = 4;
  repeated venue_configuration configs            = 5;
}
